src/LoadBalancer.js

```diff
@@ -75,6 +78,41 @@ class LoadBalancer {
     return sortedEngines[0];
   }
 
+  static weightedLoad(engines) {
+    const maxSessionsPerEngine = Config.sessionsPerEngineThreshold;
+
+    // Calculate an array containing the maximum allowed session count
+    // minus actual session count for each engine
+    const sessionsPossibleToAllocateArray = engines.map((engine) => {
+      if (!engine.engine.metrics) {
+        return 0;
+      }
+      const sessionMetric = engine.engine.metrics.filter(metric => metric.name === 'qix_active_sessions');
```

Comment:
engine.engine looks a bit strange. Perhaps better to use instance.engine or element.engine